src/Elliptic/Systems/Poisson/Equations.hpp

```diff
@@ -47,55 +51,63 @@ class er;
-                                       Tags::Coordinates<Dim, Frame::Logical>>>;
-  static void apply(
-      gsl::not_null<Scalar<DataVector>*> operator_for_field_source,
-      gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-          operator_for_auxiliary_field_source,
-      const tnsr::i<DataVector, Dim, Frame::Inertial>& grad_field,
-      const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
-      const Mesh<Dim>& mesh,
-      const InverseJacobian<DataVector, Dim, Frame::Logical, Frame::Inertial>&
-          inverse_jacobian) noexcept;
-};
+void auxiliary_fluxes(gsl::not_null<tnsr::Ij<DataVector, Dim, Frame::Inertial>*>
+                          flux_for_gradient,
+                      const Scalar<DataVector>& field) noexcept;
 
 /*!
- * \brief The interface normal dotted into the fluxes for the first order
- * formulation of the Poisson equation.
+ * \brief Compute the fluxes \f$F^i_A\f$ for the Poisson equation on a flat
+ * metric in Cartesian coordinates.
  *
- * \details For the sourced equation this is \f$-\boldsymbol{n} \cdot
- * \boldsymbol{v}(x)\f$ and for the auxiliary equation it is
- * \f$\boldsymbol{n} u\f$ (see `Poisson::FirstOrderSystem` and `dg::lift_flux`).
+ * \see Poisson::FirstOrderSystem
  */
 template <size_t Dim>
-struct ComputeFirstOrderNormalDotFluxes {
-  using argument_tags =
-      tmpl::list<LinearSolver::Tags::Operand<Field>,
-                 LinearSolver::Tags::Operand<AuxiliaryField<Dim>>,
-                 Tags::Normalized<Tags::UnnormalizedFaceNormal<Dim>>>;
+struct EuclideanFluxes {
+  using argument_tags = tmpl::list<>;
   static void apply(
-      gsl::not_null<Scalar<DataVector>*> normal_dot_flux_for_field,
-      gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
-          normal_dot_flux_for_auxiliary_field,
-      const Scalar<DataVector>& field,
-      const tnsr::I<DataVector, Dim, Frame::Inertial>& auxiliary_field,
+      const gsl::not_null<tnsr::I<DataVector, Dim, Frame::Inertial>*>
+          flux_for_field,
       const tnsr::i<DataVector, Dim, Frame::Inertial>&
-          interface_unit_normal) noexcept;
+          field_gradient) noexcept {
+    euclidean_fluxes(flux_for_field, field_gradient);
+  }
+  static void apply(
+      const gsl::not_null<tnsr::Ij<DataVector, Dim, Frame::Inertial>*>
+          flux_for_gradient,
+      const Scalar<DataVector>& field) noexcept {
+    auxiliary_fluxes(flux_for_gradient, field);
+  }
+  // clang-tidy: no runtime references
+  void pup(PUP::er& /*p*/) noexcept {}  // NOLINT
+};
+
+/*!
+ * \brief Compute the sources \f$S_A\f$ for the Poisson equation.
+ *
+ * \see Poisson::FirstOrderSystem
+ */
+struct Sources {
+  using argument_tags = tmpl::list<>;
+  static void apply(
+      const gsl::not_null<Scalar<DataVector>*> source_for_field,
+      const Scalar<DataVector>& field) noexcept {
+    *source_for_field = make_with_value<Scalar<DataVector>>(field, 0.);
```

Comment:
Only resize if necessary.

Reply:
same here, it zeroes the source

Reply:
This one can just use `get(...) = 0.` since it's a scalar.  (After resizing if necessary.)